File: RootTools2/src/main/java/com/rarnu/tools/neo/xposed/ads/FuckMms.java
package com.rarnu.tools.neo.xposed.ads;

import android.content.Context;
import com.rarnu.tools.neo.xposed.XpUtils;
import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

/**
 * Created by rarnu on 11/18/16.
 */
public class FuckMms {

    public static void fuckMms(XC_LoadPackage.LoadPackageParam loadPackageParam) {
        XpUtils.INSTANCE.findAndHookMethod("com.android.mms.ui.MessageUtils", loadPackageParam.classLoader, "isMessagingTemplateAllowed", Context.class, new XC_MethodHook() {
            protected void beforeHookedMethod(MethodHookParam paramAnonymousMethodHookParam) throws Throwable {
                Context mc = (Context) paramAnonymousMethodHookParam.args[0];
                paramAnonymousMethodHookParam.setResult(!mc.getClass().getName().toLowerCase().contains("app"));
            }
        });
        XpUtils.INSTANCE.findAndHookMethod("com.android.mms.ui.SingleRecipientConversationActivity", loadPackageParam.classLoader, "showMenuMode", new XC_MethodReplacement() {
            @Override
            protected Object replaceHookedMethod(MethodHookParam param) throws Throwable {
                return null;
            }
        });
        XpUtils.INSTANCE.findAndHookMethod("com.android.mms.util.MiStatSdkHelper", loadPackageParam.classLoader, "recordBottomMenuShown", String.class, new XC_MethodReplacement() {
            @Override
            protected Object replaceHookedMethod(MethodHookParam param) throws Throwable {
                return null;
            }
        });
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/xposed/ads/FuckMms.kt
package com.rarnu.tools.neo.xposed.ads

import android.content.Context
import com.rarnu.xfunc.*

/**
 * Created by rarnu on 11/18/16.
 */
object FuckMms {

    fun fuckMms(pkg: XposedPkg) {
        pkg.findClass("com.android.mms.ui.MessageUtils").findMethod("isMessagingTemplateAllowed", Context::class.java).hook {
            before {
                val mc = args[0] as Context
                result = !mc.javaClass.name.toLowerCase().contains("app")
            }
        }
        pkg.findClass("com.android.mms.ui.SingleRecipientConversationActivity").findMethod("showMenuMode").hook { replace { result = null } }
        pkg.findClass("com.android.mms.util.MiStatSdkHelper").findMethod("recordBottomMenuShown", String::class.java).hook { replace { result = null } }
    }
}
